osm-fudge/amenity-example.py

```diff
@@ -0,0 +1,26 @@
+"""
+Extract all objects with an amenity tag from an osm file and list them
+with their name and position.
+"""
+import osmium as o
+import sys
+
+class AmenityListHandler(o.SimpleHandler):
+
+    def print_amenity(amenity, tags, lon, lat):
+        name = tags.get('name', '')
+        print("%f %f %-15s %s" % (lon, lat, tags['amenity'], name))
+
+    def node(self, n):
+        if 'amenity' in n.tags:
+            self.print_amenity(n.tags, n.location.lon, n.location.lat)
+
+
+if __name__ == '__main__':
+    if len(sys.argv) != 2:
+        print("Usage: python amenity-example.py <osmfile>")
+        sys.exit(-1)
```

Comment:
You can shortcut by saying

    sys.exit('Usage: ...')

:)

Reply:
Ah nice 👍

Reply:
Also, should this be considered as successful exit (Status code 0) or an exit with error?

Reply:
I don't think it matters a lot in our case; where it matters is if you want to e.g. call this script from a bash script and repeat when the exit code is not zero. I think it's fine what you are doing.